### ISSUE TITLE
Exact match for strings not working

### ISSUE THREAD

**Issue by karthikmurugadoss** [vaexio/vaex#228]
I'm looking to select columns (of say names) which exactly match a query. Apologies in advance if I'm missing something basic here.

```
>>> import pandas as pd
>>> import vaex
>>> df = pd.DataFrame([["A", 43], ["B", 12], ["A", 22], ["C", 53], ["D", 12]], columns=["x", "y"])
>>> df
   x   y
0  A  43
1  B  12
2  A  22
3  C  53
4  D  12
```
Converting to vaex dataframe: 
```
>>> df = vaex.from_pandas(df)
>>> df.dtypes
index            int64
x        <class 'str'>
y                int64
dtype: object
```
The following seems to work as expected:
```
>>> df.y == 12
Expression = (y == 12)
Length: 5 dtype: bool (expression)
----------------------------------
0  False
1   True
2  False
3  False
4   True
```
When I attempt the same on the other column, I get an error:
```
>>> df.x == "A"
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/vaex/scopes.py", line 86, in evaluate
    result = self[expression]
  File "/usr/local/lib/python3.5/dist-packages/vaex/scopes.py", line 133, in __getitem__
    raise KeyError("Unknown variables or column: %r" % (variable,))
KeyError: "Unknown variables or column: '(x == A)'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.5/dist-packages/vaex/expression.py", line 245, in __repr__
    return self._repr_plain_()
  File "/usr/local/lib/python3.5/dist-packages/vaex/expression.py", line 276, in _repr_plain_
    dtype = self.dtype
  File "/usr/local/lib/python3.5/dist-packages/vaex/expression.py", line 159, in dtype
    return self.ds.dtype(self.expression)
  File "/usr/local/lib/python3.5/dist-packages/vaex/dataframe.py", line 1795, in dtype
    data = self.evaluate(expression, 0, 1)
  File "/usr/local/lib/python3.5/dist-packages/vaex/dataframe.py", line 4713, in evaluate
    value = scope.evaluate(expression)
  File "/usr/local/lib/python3.5/dist-packages/vaex/scopes.py", line 92, in evaluate
    result = eval(expression, expression_namespace, self)
  File "<string>", line 1, in <module>
NameError: name 'A' is not defined
```

**Comment by karthikmurugadoss** [comment]
Just found this to work: 
```
>>> df.select("x == \'A\'")
>>> df.evaluate(df.y, selection=True)
array(['A', 'A'], dtype=object)
```
What is the best practice here?

**Comment by JovanVeljanoski** [comment]
Hi @karthikmurugadoss 

For your example,  I believe you would get the optimal performance like this:
```
>>> df.x.str.match("A")
```

I hope this helps.

**Comment by maartenbreddels** [comment]
Hi,

you're using it correctly, this is how it should work. This is a bug (missing feature) and an unclear error msg. Before, when strings were numpy arrays we were getting this 'for free', because numpy implemented this for us. Now we have to forward the `__eq__` operator (and it's cousins) to use match. I'll keep this open until fixed.
Thanks for the report!

cheers,

Maarten

**Comment by maartenbreddels** [comment]
Note to self, we need a string_cmp function in c++ for optimal performance, since match uses regex.

**Comment by JovanVeljanoski** [comment]
This is fixed via #327.
